### DISCUSSION TITLE
Modify tag value of an OSM node

### DISCUSSION THREAD

**Discussion by wipfli** [onthegomap/planetiler#873] [Q&A]
What is the best way to modify a tag value of a given OSM node after reading the source pbf?

I want to for example change `place=state` to `place=country` in the node of French Guyana https://www.openstreetmap.org/node/313753808.

**Comment by msbarry** [comment]
Is this in the openmaptiles profile or a custom one? There's no automated way to override OSM data prior to processing it, but if you're working in code you could do something like this:

```java
    var feature = features.point("place")
      ...
    if (element.source() instanceof OsmSourceFeature source && source.originalElement().id() == 313753808L) {
      feature.setAttr("place", "country");
    }
```

Might be a nice capability to be to be able to process a osm pbf plus a changeset xml file with overrides/removals/extra features?

**Reply by wipfli** [reply]
Thanks for the reply @msbarry. This is roughly what I have done so far, assuming that this code is in the `processFeature` function.

I think it would be a nice addition to have the ability to apply a changeset xml file on a planet file. Do you know if there are existing java packages for this?

**Reply by msbarry** [reply]
You could also do a pre-processing step with osmium that applies the a changeset to the original PBF, but that would probably take a while. All of planetiler's pbf handling is custom, so the change handling would need to be custom too so I can't use an off the shelf one.  I did some prototyping in #119 that shows it would be doable, but a bit complex to handle very large change sets without loading the whole thing in memory, although it could be much simpler for small ones if we assume they fit in memory.
